cmd/go-clang-generate/generate/enum.go

```diff
@@ -156,77 +143,80 @@ func (e *Enum) addEnumStringMethods() error {
-	switch {{$.Receiver}} {
-		{{range $en := $.Cases}}case {{range $sn := $en.CaseStr}}{{$sn}}{{end}}: return "{{$en.PrettyStr}}"
-		{{end}}
-	}
+func (e *Enum) addEnumSpellingMethod() error {
+	f := newFunction("Spelling", e.Name, "", "", Type{GoName: "string"})
+	fa := newASTFunc(f)
 
-	return fmt.Sprintf("{{$.ReceiverType}} unkown %d", int({{$.Receiver}}))
-}
-`))
+	fa.generateReceiver()
 
-func (e *Enum) addEnumSpellingMethod() error {
-	var b bytes.Buffer
+	fa.addReturnType("", Type{
+		GoName: "string",
+	})
 
-	type Case struct {
-		CaseStr   []string
-		PrettyStr string
-	}
+	switchStmt := doSwitchStmt(&ast.Ident{Name: f.Receiver.Name})
+	fa.Body.List = append(fa.Body.List, switchStmt)
 
-	type Switch struct {
-		Receiver     string
-		ReceiverType string
-		Cases        []Case
-	}
+	m := make(map[uint64]*ast.CaseClause)
 
-	s := &Switch{
-		Receiver:     e.Receiver.Name,
-		ReceiverType: e.Receiver.Type.GoName,
-		Cases:        []Case{},
-	}
+	for _, enumerator := range e.Items {
+		if caseClause, ok := m[enumerator.Value]; !ok {
+			c := []ast.Expr{&ast.Ident{Name: enumerator.Name}}
 
-	m := make(map[string]struct{})
+			ret := &ast.ReturnStmt{
+				Results: []ast.Expr{},
+			}
 
-	for i, enumerator := range e.Items {
-		if _, ok := m[enumerator.Name]; ok {
-			continue
-		}
+			ret.Results = append(ret.Results, doStringLit(strings.Replace(enumerator.Name, "_", "=", 1)))
 
-		c := Case{
-			CaseStr:   []string{enumerator.Name},
-			PrettyStr: strings.Replace(enumerator.Name, "_", "=", 1),
-		}
-		m[enumerator.Name] = struct{}{}
+			b := []ast.Stmt{ret}
 
-		for s := i + 1; s < len(e.Items); s++ {
-			if e.Items[s].Value == enumerator.Value {
-				c.CaseStr = append(c.CaseStr, "/*"+e.Items[s].Name+"*/")
-				c.PrettyStr = c.PrettyStr + ", " + e.Items[s].Name[strings.Index(e.Items[s].Name, "_")+1:]
-				m[e.Items[s].Name] = struct{}{}
-			}
+			caseClause = doCaseClause(c, b)
+			switchStmt.Body.List = append(switchStmt.Body.List, caseClause)
+			m[enumerator.Value] = caseClause
+		} else {
+			retStr := caseClause.Body[0].(*ast.ReturnStmt).Results[0].(*ast.BasicLit).Value
+			retStr = retStr[0:len(retStr)-1] + ", " + enumerator.Name[strings.Index(enumerator.Name, "_")+1:] + "\""
+			caseClause.Body[0].(*ast.ReturnStmt).Results[0].(*ast.BasicLit).Value = retStr
 		}
-
-		s.Cases = append(s.Cases, c)
 	}
 
-	if err := templateGenerateEnumSpelling.Execute(&b, s); err != nil {
-		return err
-	}
+	fa.addReturnItem(doCall(
+		"fmt",
+		"Sprintf",
+		doStringLit(f.Receiver.Type.GoName+" unkown %d"),
+		doCast("int", &ast.Ident{Name: f.Receiver.Name}),
+	))
+
+	fa.addEmptyLine()
+	fa.addStatement(fa.ret)
 
-	e.Methods = append(e.Methods, b.String())
+	e.Methods = append(e.Methods, generateFunctionString(fa))
 
 	return nil
 }
 
-func (e *Enum) addEnumMethod(tmpl *template.Template) error {
-	var b bytes.Buffer
-	if err := tmpl.Execute(&b, e); err != nil {
-		return err
+func (e *Enum) addEnumMethod(fName string) error {
```

Comment:
This function now just makes aliases for the Spelling method. Please redo the signature to "addSpellingMethodAlias(name string) error"